app/src/pages/inside/testCaseLibraryPage/addTestCasesToTestPlanModal/addTestCasesToTestPlanModal.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/*
+ * Copyright 2025 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from 'classnames/bind';
+import { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { InjectedFormProps, reduxForm } from 'redux-form';
+import { useIntl } from 'react-intl';
+
+import { Modal } from '@reportportal/ui-kit';
+
+import { AsyncAutocomplete } from 'componentLibrary/autocompletes/asyncAutocomplete';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { commonValidators } from 'common/utils/validation';
+import { hideModalAction, withModal } from 'controllers/modal';
+import { LoadingSubmitButton } from 'components/loadingSubmitButton';
+import { messages } from './messages';
+import { projectKeySelector } from 'controllers/project';
+import { TestPlanDto } from 'controllers/testPlan';
+import { URLS } from 'common/urls';
+
+import { useAddTestCasesToTestPlan } from './useAddTestCasesToTestPlan';
+
+import styles from './addTestCasesToTestPlanModal.module.scss';
+import { AddTestCasesToTestPlanFormData, AddTestCasesToTestPlanModalProps } from './types';
+
+const cx = classNames.bind(styles) as typeof classNames;
+
+export const ADD_TO_TEST_PLAN_MODAL_KEY = 'addToTestPlanModalKey';
+export const ADD_TO_TEST_PLAN_MODAL_FORM = 'add-to-test-plan-form';
+export const SELECTED_TEST_PLAN_FIELD_NAME = 'selectedTestPlan';
+
+export const AddTestCasesToTestPlanModal = ({
+  change,
+  handleSubmit,
+  data,
+}: AddTestCasesToTestPlanModalProps &
+  InjectedFormProps<AddTestCasesToTestPlanFormData, AddTestCasesToTestPlanModalProps>) => {
+  const { selectedTestCaseIds } = data;
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+
+  const projectKey = useSelector(projectKeySelector);
+
+  const selectedTestCasesLength = selectedTestCaseIds?.length || 0;
```

Comment:
Let's use size from `es-toolkit`
```suggestion
  const selectedTestCasesLength = size(selectedTestCaseIds);
```